### ISSUE TITLE
ReadStream errors aren't being listened

### ISSUE THREAD

**Issue by scriptype** [form-data/form-data#180]
When trying to read a file which is used by other processes or is busy/locked for some reason, I'm getting this error on a Windows machine (same case seems not to be problematic on os x):

```
Error: EBUSY: resource busy or locked, open [file-path-here] at Error (native)
```

form-data module, in my case, is running inside [superagent](https://github.com/visionmedia/superagent) module which I'm using for handling http stuff in node. So checking whether a specific file exists or is okay to read is beyond the scope of my module's logic, instead I'm expecting low-level modules (i.e. superagent > form-data > combined-stream) to handle any low-level fs errors for me.

I've searched for similar issues and found: #165, #59 and #58, which can relate to this issue.

I've also inspected the code a little and found that there are no listeners bound to error event of `CombinedStream`. I think a simple addition to constructor like:

```
this.on('error', this._error.bind(this))
```

would solve the problem, as far as I can see. I can make the changes and open a PR.

**Comment by alexindigo** [comment]
Actually, it's the other way around. Libs don't have the knowledge of what is expected to do when such errors occur. So it's your files and your app's responsibility and understanding of what to do in case of an errors.

Let's brainstorm a little, for example that kind of errors handled somewhere inside the libs you are using. What action supposed to be performed if such error occurred, to make sense for your app?

Thank you.

**Comment by scriptype** [comment]
Hello @alexindigo, thanks for the quick reply. I agree that handling logic should be done in consumer side. The thing is, form-data isn't handling/catching these errors and process stops because of this.

form-data should've been listening for errors and emitting them as they are, without doing any opinionated operation on them. So any consumer could listen for errors and do consumer-specific logic.

Actually I'm seeing this exact thing is done while wrapping `request` in submit function. When a user of library wants to handle errors that happen during submission, library provides the error for them as the first parameter of the callback. This isn't done for reading files, however.

A basic example:
```js
require('fs')
  .createReadStream('someProblematicFile')
  .on('error', (e) => { myBus.emit('error', e) })
```

Without the error-listening part, this script would cause running process to exit non-zero (it's doing so in the case of `Error: EBUSY`), as an error would be thrown and we wouldn't be catching it.

**Comment by alexindigo** [comment]
I see. It does make more sense yes. Thanks for the explanation. 

Since FormData is creating stream out of itself, you should be able to listen for the `error` event on your FormData instance.

and this is what CombinedStream does

```
CombinedStream.prototype._handleErrors = function(stream) {
  var self = this;
  stream.on('error', function(err) {
    self._emitError(err);
  });
};

CombinedStream.prototype._emitError = function(err) {
  this._reset();
  this.emit('error', err);
};
```

Which looks like something you're describing.

Please let me if it helps.

**Comment by scriptype** [comment]
Thanks for the clarification.

Apparently we can directly listen our FormData instance for errors.

My issue doesn't seem to be related to here. Thanks again.

**Comment by niftylettuce** [comment]
Fixed in superagent v7.0.0.

See <https://github.com/visionmedia/superagent/releases/tag/v7.0.0> for more insight.

We've also published this deprecation notice:

```sh
Please upgrade to v7.0.0+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>. Thanks to @shadowgate15 (aka Super Agent Taylor), @spence-s, and @niftylettuce.
```
